src/lib/classes/access-control-list.js

```diff
@@ -0,0 +1,333 @@
+  };
+
+  /**
+   * @function getACLFile Retrieves the acl file from the network
+   * @returns {Object} Acl fetched file
+   */
+  getACLFile = async () => {
+    try {
+      const result = await solid.fetch(this.aclUri);
+      if (result.status === 403) throw new Error(result.message || 'Not authorized');
+      if (result.status === 404) return this.getParentACL(this.documentUri);
+      return result;
+    } catch (e) {
+      throw e;
+    }
+  };
+
+  /**
+   * @function getSubjects Creates an object based on a ldflex proxy document
+   * @param {Proxy} document The base ldflex proxy document from where the data will be extracted
+   * @returns {Array<Permissions>} A custom object to visualized the acls of a document or container it a Pod
+   */
+  getSubjects = async document => {
+    let subjects = [];
+    for await (const subject of document.subjects) {
+      let agents = [];
+      let modes = [];
+      for await (const agent of subject['acl:agent']) {
+        agents = [...agents, agent.value];
+      }
+      for await (const mode of subject['acl:mode']) {
+        const modeName = mode.value ? mode.value.split('#')[1] : '';
+        modes = [...modes, modeName];
+      }
+      const agentClass = await subject['acl:agentClass'];
+      if (agents.length > 0) subjects = [...subjects, { subject, agents, modes }];
+      if (agentClass) subjects = [...subjects, { subject, agents: null, modes }];
+    }
+    return subjects;
+  };
+
+  /**
+   * @function getPermissions Retrieves the acl file as an array of Permissions objects
+   * @returns {Array<Permissions>} An array of permissions
+   */
+  getPermissions = async () => {
+    try {
+      if (!this.acl) {
+        const file = await this.getACLFile();
+        if (!file) throw new Error('ACL File was not found for the resource');
+        const doc = await ldflex[file.url];
+        const permissions = this.getSubjects(doc);
+        this.setAcl = permissions;
+        return permissions;
+      }
+      return this.acl;
+    } catch (e) {
+      throw e;
+    }
+  };
+
+  /**
+   * @function deleteACL Deletes the entire acl file, leaving the default acls from the parent container
+   * @return {Boolean} Returns if deletion was successful
+   */
+  deleteACL = async () => {
+    const result = await solid.fetch(this.aclUri, { method: 'DELETE' });
+    return result.ok;
+  };
+
+  /**
+   * @function isSameMode checks if two sorted arrays of modes are equal
+   * @param {Array<String>} modes1 An array of mode names to be compared
+   * @param {Array<String>} modes2 An array of mode names to be compared
+   * @returns {Boolean} The result of comparing two array to see if are equals
+   */
+  isSameMode = (modes1, modes2) => isEqual(modes1.sort(), modes2.sort());
+
+  /**
+   * @function createMode creates a new mode in the acl file
+   * @param {Array<Permissions>} permissions An array of permissions with the necessary modes and agents
```

Comment:
Params seem to have changed since the comment was written, should have modes and agents

Reply:
Yes should be only Permission not Array

Reply:
done